scripts/AI/States/Pursue.gd

```diff
@@ -0,0 +1,71 @@
+extends AIState
+class_name Pursue
+
+var _last_target_pos: Vector2
+
+var target_move_direction: Vector2
+var find_timer = Timer.new()
+var wait_rotation_timer = Timer.new()
+
+var tween: Tween
+
+var searching: bool
+
+func _ready():
+	await owner.ready
+
+	find_timer.timeout.connect(_on_pursuit_time_timeout)
+	find_timer.wait_time = state_machine._enemy.pursue_find_time
+	find_timer.one_shot = true
+	add_child(find_timer)
+	
+	wait_rotation_timer.timeout.connect(_on_wait_point_timeout)
+	wait_rotation_timer.wait_time = state_machine._enemy.pursue_wait_rotation_time
+	wait_rotation_timer.one_shot = true
+	add_child(wait_rotation_timer)
+
+func update(_delta: float) -> void:
+	if state_machine._enemy.global_transform.origin.distance_to(_last_target_pos) <= 20 and not searching:
+		searching = true
+		find_timer.start()
+		set_movement_target(state_machine._enemy.global_transform.origin + target_move_direction * 500)
```

Comment:
*500 is the distance he will look for the player ? the one he is going to walk for ?
If it is, it should be a parameter please.